src/hotspot/share/gc/shared/referenceProcessor.cpp

```diff
@@ -633,60 +632,58 @@ void ReferenceProcessor::balance_queues(DiscoveredList ref_lists[])
   size_t avg_refs = total_refs / _num_queues + 1;
   uint to_idx = 0;
   for (uint from_idx = 0; from_idx < _max_num_queues; from_idx++) {
-    bool move_all = false;
-    if (from_idx >= _num_queues) {
-      move_all = ref_lists[from_idx].length() > 0;
+    size_t from_len = ref_lists[from_idx].length();
+    if (from_len == 0) {
+      // empty
+      continue;
```

Comment:
Kind of waffling over this, but I'm not sure this is worthwhile. The result falls out of the 
`remaining_to_move` calculation and use, and it's kind of cluttering.

Reply:
I thought it's clearer to handle the empty-case explicitly. No strong feelings either way; removed.